web/client-api/src/main/java/io/deephaven/web/client/state/ClientTableState.java

```diff
@@ -233,6 +233,90 @@ public TableTicket getHandle() {
+                return ChunkType.Object;
+            }
+            if (dataType == Long.class || dataType == long.class) {
+                // JS client holds longs as LongWrappers
+                return ChunkType.Object;
+            }
+            return ChunkType.fromElementType(dataType);
+        }).toArray(ChunkType[]::new);
+    }
+
+    /**
+     * Returns the Java Class to represent each column in the table. This lets the client replace certain JVM-only
+     * classes with alternative implementations, but still use the simple
+     * {@link io.deephaven.extensions.barrage.chunk.ChunkReader.TypeInfo} wrapper.
+     */
+    public Class<?>[] columnTypes() {
+        return Arrays.stream(tableDef.getColumns())
+                .map(ColumnDefinition::getType)
+                .map(t -> {
+                    switch (t) {
+                        case "boolean":
+                        case "java.lang.Boolean":
+                            return boolean.class;
+                        case "char":
+                        case "java.lang.Character":
+                            return char.class;
+                        case "byte":
+                        case "java.lang.Byte":
+                            return byte.class;
+                        case "int":
+                        case "java.lang.Integer":
+                            return int.class;
+                        case "short":
+                        case "java.lang.Short":
+                            return short.class;
+                        case "long":
+                        case "java.lang.Long":
+                            return long.class;
+                        case "java.lang.Float":
+                        case "float":
+                            return float.class;
+                        case "java.lang.Double":
+                        case "double":
+                            return double.class;
+                        case "java.time.Instant":
+                            return DateWrapper.class;
+                        case "java.math.BigInteger":
+                            return BigIntegerWrapper.class;
+                        case "java.math.BigDecimal":
+                            return BigDecimalWrapper.class;
+                        default:
+                            return Object.class;
+                    }
+                })
+                .toArray(Class<?>[]::new);
+    }
+
+    /**
+     * Returns the Java Class to represent the component type in any list/array type. Only used to detect if a byte[]
+     * should be used for holding data at this time.
+     */
+    public Class<?>[] componentTypes() {
+        // The only componentType that matters is byte.class
+        return Arrays.stream(tableDef.getColumns()).map(ColumnDefinition::getType).map(t -> {
+            if (!t.endsWith("[]")) {
+                return null;
+            }
+            if (t.equals("io.deephaven.vector.ByteVector[]")) {
```

Comment:
Agreed, that doesn't make sense - I think I tried to work forward and backwards (consumer and producer) and found the byte case was the only one to matter, but I think it still doesn't quite apply here.

Re-reading, I'm seeing that AbstractColumnSource's constructor can assign componentType either from a vector subtype, or from reading the array's type - but looking at the consuming side, only the VarBinaryChunkInputStreamGenerator vs VarListChunkInputStreamGenerator check in DefaultChunkInputStreamGeneratorFactory actually minds what the value of componentType is (disregarding VectorChunkInputStreamGenerator, since the client can't support that class).

The current WebChunkReaderFactory doesn't appear to care about this at all though, so I'll rewrite and add a note that #2102 and #5907 should follow up on this better.